### ISSUE TITLE
[Issue]: Directory watch disabled if subdirectory (e.g. lost+found) is unreadable

### ISSUE THREAD

**Issue by niclan** [jellyfin/jellyfin#7130]
### Please describe your bug

This morning I've reviewed my jellyfin logs for the first time.  I've been able to correct some permission issues and get them quite quiet. But this remains:

```
jan. 07 09:40:15 gont jellyfin[3320044]: System.UnauthorizedAccessException: Access to the path '/space/dvds/lost+found' is denied.
jan. 07 09:40:15 gont jellyfin[3320044]:  ---> System.IO.IOException: Permission denied
jan. 07 09:40:15 gont jellyfin[3320044]:    --- End of inner exception stack trace ---
jan. 07 09:40:15 gont jellyfin[3320044]: [09:40:15] [INF] Stopping directory watching for path /space/dvds
```

The lost+found directory should not be readable to non-admins and should in any case not be catalogued. It seems unnecessary to let a permission denied exception disable watching a directory hierarchy.

Regards,
  Nicolai

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Linux, Ubuntu 20.04.3 LTS
- Virtualization: Running directly on host, not in VM or container
- Clients: This is a storage backend issue
- Browser: This is a storage backend issue
- FFmpeg Version: This is a storage backend issue
- Playback Method: This is a storage backend issue
- Hardware Acceleration: This is a storage backend issue
- Plugins: This is a storage backend issue
- Reverse Proxy: This is a storage backend issue
- Base URL: This is a storage backend issue
- Networking: This is a storage backend issue
- Storage: Local disk, ext4 filesystem
```

### Jellyfin logs

```shell
jan. 07 09:40:15 gont jellyfin[3320044]: [09:40:15] [ERR] Error in Directory watcher for: /space/dvds
jan. 07 09:40:15 gont jellyfin[3320044]: System.UnauthorizedAccessException: Access to the path '/space/dvds/lost+found' is denied.
jan. 07 09:40:15 gont jellyfin[3320044]:  ---> System.IO.IOException: Permission denied
jan. 07 09:40:15 gont jellyfin[3320044]:    --- End of inner exception stack trace ---
jan. 07 09:40:15 gont jellyfin[3320044]: [09:40:15] [INF] Stopping directory watching for path /space/dvds
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by niclan** [comment]
Ticket ping to prevent closing. This is a real bug.

**Comment by AinoraZ** [comment]
Can confirm that described bug is still present in jellyfin 10.8.5.

**Comment by niclan** [comment]
Updated to jellyfin 10.8.10-1 this morning.

```
2023-04-27 08:00:15.637 +02:00] [ERR] Error in Directory watcher for: "/space/dvds"
System.UnauthorizedAccessException: Access to the path '/space/dvds/lost+found' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
[2023-04-27 08:00:15.637 +02:00] [INF] Stopping directory watching for path "/space/dvds"
```

**Comment by nielsvanvelzen** [comment]
The lost+found directory is already in the ignore patterns (https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Library/IgnorePatterns.cs#L50-L51). Please share additional information like the directory structure and full logs (not just a small snippet) and maybe someone can figure out what goes wrong.

**Comment by niclan** [comment]
@nielsvanvelzen - that was all that was logged about the watch of my dvds directory. 

I guess I need to tweak the log level somewhere?  Since I'm a programmer I might also be able to squint at the code and test something.

**Comment by szotsaki** [comment]
Still in issue with 10.8.13.

Exception trace:
```
[ERR] [23] Emby.Server.Implementations.IO.LibraryMonitor: Error in Directory watcher for: "/data/movies"                                                      
System.UnauthorizedAccessException: Access to the path '/data/movies/External/lost+found' is denied.                                                                                           
 ---> System.IO.IOException: Permission denied                                                                                                                                                 
   --- End of inner exception stack trace --- 
```

**Comment by evert** [comment]
Just to add another data point:

I switched to Jellyfin about a year ago, but new files not being detected was a big usability issue. I'm pretty familiar with linux admin, so I tried to dig in from time to time to no avail and worked around it 

Today I got relentless, kept the jellyfin log open, googling variations (maybe it's inotify, maybe it's btrfs?) until I landed on #7724. I did see a permissions error deep in my music folder on `LCD Soundsystem` but didn't assume it would kill the watcher. After fixing permissions, it worked perfectly!

```
[2024-06-03 00:28:51.111 -04:00] [ERR] Error in Directory watcher for: "/media/storage/music"
System.UnauthorizedAccessException: Access to the path '/media/storage/music/albums/LCD Soundsystem' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
```

Would love to see a fix for this, but if it's hard or hard to prioritize this I'd love to make a smaller suggestion: if the watcher process is about to crash due to an uncaught exception, a log message explicitly mentioning this would probably have led me to this solution much quicker!

The error was there, but I didn't intuit the implication.

Thanks for Jellyfin! It's awesome. As an Open Source dev for almost 23 years I know it can be thankless. You are appreciated!
